api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -219,33 +212,32 @@ public function patchOne($id, $childId, Request $request)
     public function patchMany($id, Request $request)
     {
         $requestCollection = $request->data;
+        $model = $this->findParentEntity($id);
+        $childModels = $this->findOrFailChildrenCollection($requestCollection, $model);
 
-        $model = $this->getParentModel($id);
-
-        $relation = $this->getRelation($model);
-        $childKey = $this->getChildKey($relation);
-
-        $ids = $this->getIds($requestCollection, $childKey, $this->validateChildRequestRule);
-        $childModels = [];
-        if (!empty($ids)) {
-            $relation->getBaseQuery()->whereIn($childKey, $ids);
-            $childModels = $relation->getResults();
-        }
-
-        if (!empty($childModels) && $childModels->count() !== count($ids)) {
-            throw $this->notFoundManyException($ids, $childModels, $childKey);
-        }
-
+        $error = false;
+        $errors = [];
         foreach ($requestCollection as $requestEntity) {
-            $id = $requestEntity[$childKey];
+            $id = $this->getIdOrNull($requestEntity, $this->getChildModel()->getKeyName());
```

Comment:
The validation part of this needs to be extracted out to a separate `validateCollectionRequest()` method, as it will be common to both child and parent entities, and should be decoupled from model instantiation.